### ISSUE TITLE
Workload User Group Creation Fail in Resource Manager

### ISSUE THREAD

**Issue by TheKoguryo** [oracle-quickstart/oci-landing-zones#84]
When apply templates/enterprise-landing-zone or templates/elz-workload, I can see that the final state is Succeeded.
But user groups for workload are not created. And I can find the related error logs in Resource Manager logs.

```
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0](local-exec): Executing: ["/bin/sh" "-c" "python3 scripts/manage_identity_domain.py -d ocid1.domain.oc1..aaaaaaaax______s6q -g OCI-ELZ-UGP-N-APP-ADMIN OCI-ELZ-UGP-N-DB-ADMIN OCI-ELZ-UGP-N-WRK-ADMIN"]
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0] (local-exec): Traceback (most recent call last):
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0](local-exec):   File "scripts/manage_identity_domain.py", line 127, in <module>
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0] (local-exec):     manage_id = ManageIdentityDomain(args.domain_id, args.group_names)
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0](local-exec):   File "scripts/manage_identity_domain.py", line 14, in __init__
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0] (local-exec):     self.config, self.auth = self.set_up_oci_config()
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0](local-exec):   File "scripts/manage_identity_domain.py", line 29, in set_up_oci_config
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0] (local-exec):     user=config['user'],
module.nonprod_environment.module.workload.module.groups.null_resource.groups[0] (local-exec): KeyError: 'user'
```

I add a new line to print the config value.

modules/identity-domain-group/scripts/manage_identity_domain.py
```
26            config = oci.config.from_file()
27            print("config: " + str(config))
28            auth = oci.Signer(
```

Logs in Resource Manager
```
module.groups.null_resource.groups[0] (local-exec): config: {'log_requests': False, 'additional_user_agent': '', 'pass_phrase': None, 'tenancy': 'ocid1.tenancy.oc1..aaaaaaaac_____ajq', 'region': 'ap-seoul-1', 'delegation_token_file': '/home/orm/delegation_token'}
```

In modules/identity-domain-group/scripts/manage_identity_domain.py, the codes use private_key to get an auth.
But to run OELZ v2 in Resource Manager, it is required to use a delegation_token_file to fix this user group creation fail.

**Comment by VinayKumar611** [comment]
Hi @TheKoguryo , 

Thanks for reporting this issue . We had removed the Python dependencies from the Workload Stack code and you can find the fix in this branch. Please download this file templates/elz-workload/iam.tf

Please download this file templates/elz-workload/iam.tf from this branch https://github.com/oracle-quickstart/oci-landing-zones/tree/Workload_Groups .

On next release cycle , we will merge the fix on the release branch and update you. 

PS : This is Alpha Release Branch , so no testing has been done . Will be prone to some unwanted bugs.

**Comment by VinayKumar611** [comment]
Fixed the issue on latest release v2.2.0.
